classes/UpgradeTools/Module/Source/Provider/ComposerSourceProvider.php

```diff
@@ -0,0 +1,103 @@
+<?php
+
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Academic Free License 3.0 (AFL-3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/AFL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/AFL-3.0 Academic Free License 3.0 (AFL-3.0)
+ */
+
+namespace PrestaShop\Module\AutoUpgrade\UpgradeTools\Module\Source\Provider;
+
+use PrestaShop\Module\AutoUpgrade\Parameters\FileConfigurationStorage;
+use PrestaShop\Module\AutoUpgrade\Parameters\UpgradeFileNames;
+use PrestaShop\Module\AutoUpgrade\Services\ComposerService;
+use PrestaShop\Module\AutoUpgrade\UpgradeTools\Module\Source\ModuleSource;
+
+/*
+ * Gets the modules bundled with a PrestaShop release by reading its composer.lock file.
+ */
+class ComposerSourceProvider extends AbstractModuleSourceProvider
+{
+    /** @var string */
+    private $prestaShopReleaseFolder;
+
+    /** @var ComposerService */
+    private $composerService;
+
+    /** @var FileConfigurationStorage */
+    private $fileConfigurationStorage;
+
+    public function __construct(string $prestaShopReleaseFolder, ComposerService $composerService, FileConfigurationStorage $fileConfigurationStorage)
+    {
+        $this->prestaShopReleaseFolder = $prestaShopReleaseFolder;
+        $this->composerService = $composerService;
+        $this->fileConfigurationStorage = $fileConfigurationStorage;
+    }
+
+    /** {@inheritdoc} */
+    public function getUpdatesOfModule(string $moduleName, string $currentVersion): array
+    {
+        if (null === $this->localModuleZips) {
+            $this->warmUp();
+        }
+
+        $sources = [];
+
+        foreach ($this->localModuleZips as $zip) {
+            if ($zip->getName() !== $moduleName) {
+                continue;
+            }
+
+            if (version_compare($zip->getNewVersion(), $currentVersion, '<=')) {
+                continue;
+            }
+
+            $sources[] = $zip;
+        }
```

Comment:
```suggestion
foreach ($this->localModuleZips as $zip) {
    if ($zip->getName() === $moduleName && version_compare($zip->getNewVersion(), $currentVersion, '>')) {
        $sources[] = $zip;
    }
}
```

Reply:
This methods should actually be deleted in this file, because the parent class provides it